### ISSUE TITLE
About re-trained HMR 2.0

### ISSUE THREAD

**Issue by tszhang97** [saidwivedi/TokenHMR#6]
![111](https://github.com/saidwivedi/TokenHMR/assets/39046939/33a3dc0a-7384-484a-aa28-2a2ce82b5b39)
Can you share the retrained  HMR 2.0b model?
By the way, I compared some qualitative results and found that hmr 2.0 performance is better.
tokenhmr
![image](https://github.com/saidwivedi/TokenHMR/assets/39046939/0b203c01-663d-445c-84c6-6fa63ed735e6)
hmr 2.0
![image](https://github.com/saidwivedi/TokenHMR/assets/39046939/5520e627-6bfb-418c-ab67-5d3d4da00277)
tokenhmr
![image](https://github.com/saidwivedi/TokenHMR/assets/39046939/f5720afb-7d47-42dd-a407-5915878bbb90)
hmr 2.0 
![image](https://github.com/saidwivedi/TokenHMR/assets/39046939/8c3ce88d-5a5d-416d-89a9-a458a7c7e6bc)

**Comment by saidwivedi** [comment]
Thank you for trying out the demo. [Here](https://download.is.tue.mpg.de/download.php?domain=tokenhmr&sfile=hmr2_bedlam.ckpt) is the retrained HMR2.0b model with BEDLAM as additional training data.

In the paper, we argue that models that overfit to 2D keypoints or pseudo-GT (hence better image alignment) suffer from drop in 3D accuracy (see the MVE of TokenHMR and HMR2.0 on EMDB and 3DPW datasets). Hence, TokenHMR will have lower image alignment than HMR2.0.

However, we have a new model (see [tokenhmr_v2](https://download.is.tue.mpg.de/download.php?domain=tokenhmr&sfile=tokenhmr_v2.ckpt)) which has better image alignment than the TokenHMR model used for the paper. Feel free to try that. Hope it answers your question.

**Comment by tszhang97** [comment]
Thanks for your reply, I will try the new model.

**Comment by tszhang97** [comment]
@saidwivedi I use this code to merge new HMR2.0b  model to the origin model.
`    
    import torch
    hmr_old_path = "xxx/hmr2.ckpt"
    hmr_new_path = "xxx/hmr2_bedlam.ckpt"
    m1 = torch.load(hmr_old_path)
    m2 = torch.load(hmr_new_path)
    for k, v in m2["state_dict"].items():
        if k in m1["state_dict"]:
            m1["state_dict"][k] = v
        else:
            print(k, "not in old model")

    save_path = "xxx/hmr2_new.ckpt"
    torch.save(m1, save_path)
`
Then, I use hmr2_new.ckpt replacing the model file used in 4D-Humans. The results are wired, like this
![image](https://github.com/saidwivedi/TokenHMR/assets/39046939/eb41f718-6841-4210-b3f8-a665971d9a16)

**Comment by saidwivedi** [comment]
Sorry for the delayed response because of CVPR. 

You can simply run the baseline by changing the config field (see [here](https://github.com/saidwivedi/TokenHMR/blob/e7288b65b83895261aaa64a7ee29d5cc3876eb53/tokenhmr/lib/configs_hydra/experiment/tokenhmr_release.yaml#L65)) to `transformer_decoder`.

Are you using the original HMR2.0 repo to run the TokenHMR baseline model (HMR2.0b-BEDLAM)? If so, then it is not compatible because of different implementation of `rot6d_to_rotmat` function (see [here](https://github.com/saidwivedi/TokenHMR/blob/e7288b65b83895261aaa64a7ee29d5cc3876eb53/tokenhmr/lib/utils/geometry.py#L83)).

Hope it helps :)
